File: app/src/main/java/com/android/stanlyfang/umengshare/sharetools/SharePlatform.java
package com.android.stanlyfang.umengshare.sharetools;

import android.support.annotation.DrawableRes;
import android.support.annotation.IntDef;


import com.android.stanlyfang.umengshare.R;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * @ProjectName: MultiFunctionAndroidDemo
 * @PackageName: com.umeng.soexample.share.utils.sharetools
 * @Author StanlyFang
 * @Version V1.0
 * @Date 2018/9/21 11:21
 * @Description:
 * @Update :
 */
public class SharePlatform {

    static final int WECHAT          = 1;
    static final int QQ              = 2;
    static final int SINA            = 3;
    static final int WECHAT_CIRCLE   = 4;
    static final int QZONE           = 5;
    static final int OPEN_IN_BROWSER = 6;
    static final int COPY_URL        = 7;
    static final int COPY_TEXT       = 8;

    @IntDef({WECHAT, QQ, SINA, WECHAT_CIRCLE, QZONE, OPEN_IN_BROWSER, COPY_URL, COPY_TEXT})
    @Retention(RetentionPolicy.RUNTIME)
    @interface PlatfromEnum {
    }

    public SharePlatform(@PlatfromEnum int platform) {
        switch (platform) {
            case WECHAT:
                this.ID = WECHAT;
                this.Name = "微信";
                this.resId = R.mipmap.ic_launcher;
                break;
            case QQ:
                this.ID = QQ;
                this.Name = "QQ";
                this.resId = R.mipmap.ic_launcher;
                break;
            case SINA:
                this.ID = SINA;
                this.Name = "微博";
                this.resId = R.mipmap.ic_launcher;
                break;
            case WECHAT_CIRCLE:
                this.ID = WECHAT;
                this.Name = "朋友圈";
                this.resId = R.mipmap.ic_launcher;
                break;
            case QZONE:
                this.ID = QZONE;
                this.Name = "QQ空间";
                this.resId = R.mipmap.ic_launcher;
                break;
            case COPY_TEXT:
                this.ID = COPY_TEXT;
                this.Name = "赋值内容";
                this.resId = R.mipmap.ic_launcher;
                break;
            case COPY_URL:
                this.ID = COPY_URL;
                this.Name = "复制地址";
                this.resId = R.mipmap.ic_launcher;
                break;
            case OPEN_IN_BROWSER:
                this.ID = OPEN_IN_BROWSER;
                this.Name = "浏览器打开";
                this.resId = R.mipmap.ic_launcher;
                break;
        }
    }


    private int    ID;
    private String Name;
    private int    resId;

    public SharePlatform() {

    }

    public SharePlatform(@PlatfromEnum int ID, String Name, int resId) {
        this.ID = ID;
        this.Name = Name;
        this.resId = resId;
    }


    public int getID() {
        return ID;
    }

    public void setID(@PlatfromEnum int ID) {
        this.ID = ID;
    }

    public String getName() {
        return Name;
    }

    public void setName(String name) {
        Name = name;
    }

    public int getResId() {
        return resId;
    }

    public void setResId(@DrawableRes int resId) {
        this.resId = resId;
    }
}

File: umenghelper/src/main/java/com/stanly/umenghelper/SharePlatform.kt
package com.stanly.umenghelper

import android.support.annotation.DrawableRes
import android.support.annotation.IntDef
import com.stanly.umenghelper.enumerate.SharePlatfromEnum

import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

/**
 * @ProjectName: MultiFunctionAndroidDemo
 * @PackageName: com.umeng.soexample.share.utils.sharetools
 * @Author StanlyFang
 * @Version V1.0
 * @Date 2018/9/21 11:21
 * @Description:
 * @Update :
 */
class SharePlatform {

    var id: SharePlatfromEnum = SharePlatfromEnum.WECHAT
    var name: String? = null
    var resId: Int = 0

    constructor(ID: SharePlatfromEnum, Name: String, resId: Int) {
        this.id = ID
        this.name = Name
        this.resId = resId
    }

    constructor(platform: SharePlatfromEnum) {
        when (platform) {
            SharePlatfromEnum.WECHAT -> {
                this.id = SharePlatfromEnum.WECHAT
                this.name = "微信"
                this.resId = R.drawable.umeng_socialize_wechat
            }
            SharePlatfromEnum.QQ -> {
                this.id = SharePlatfromEnum.QQ
                this.name = "QQ"
                this.resId = R.drawable.umeng_socialize_qq
            }
            SharePlatfromEnum.SINA -> {
                this.id = SharePlatfromEnum.SINA
                this.name = "微博"
                this.resId = R.drawable.umeng_socialize_sina
            }
            SharePlatfromEnum.WECHAT_CIRCLE -> {
                this.id = SharePlatfromEnum.WECHAT_CIRCLE
                this.name = "朋友圈"
                this.resId = R.drawable.umeng_socialize_wxcircle
            }
            SharePlatfromEnum.QZONE -> {
                this.id = SharePlatfromEnum.QZONE
                this.name = "QQ空间"
                this.resId = R.drawable.umeng_socialize_qzone
            }
            SharePlatfromEnum.COPY_TEXT -> {
                this.id = SharePlatfromEnum.COPY_TEXT
                this.name = "复制内容"
                this.resId = R.drawable.umeng_socialize_copy
            }
            SharePlatfromEnum.COPY_URL -> {
                this.id = SharePlatfromEnum.COPY_URL
                this.name = "复制地址"
                this.resId = R.drawable.umeng_socialize_copyurl
            }
            SharePlatfromEnum.OPEN_IN_BROWSER -> {
                this.id = SharePlatfromEnum.OPEN_IN_BROWSER
                this.name = "浏览器打开"
                this.resId = R.drawable.ic_launcher_round
            }
        }
    }


}
